datafusion/jit/src/api.rs

```diff
@@ -604,6 +606,15 @@ impl<'a> CodeBlock<'a> {
             internal_err!("No func with the name {} exist", fn_name)
         }
     }
+
+    pub fn deref(&self, ptr: Expr, ty: JITType) -> Result<Expr> {
```

Comment:
What do you think about adding docstrings? 

```suggestion
    /// Return the value pointed to by the ptr stored in `ptr`
    pub fn deref(&self, ptr: Expr, ty: JITType) -> Result<Expr> {
```